corehq/apps/geospatial/views.py

```diff
@@ -175,3 +186,82 @@ def post(self, request, *args, **kwargs):
         return json_response({
             'status': 'success'
         })
+
+
+@require_GET
+@login_and_domain_required
+def get_paginated_cases_or_users_without_gps(request, domain):
+    page = int(request.GET.get('page', 1))
+    limit = int(request.GET.get('limit', 5))
+    case_or_user = request.GET.get('data_type', 'case')
+    query = request.GET.get('query', '')
+    assert page > 0, 'Page must be a positive number'
+    assert limit > 0, 'Limit must be a positive number'
+    assert case_or_user == 'case' or case_or_user == 'user', 'Data type must be "case" or "user"'
+
+    if case_or_user == 'user':
+        data = _get_paginated_users_without_gps(domain, page, limit, query)
+    else:
+        data = _get_paginated_cases_without_gps(domain, page, limit, query)
+    return JsonResponse(data)
+
+
+def _get_paginated_cases_without_gps(domain, page, limit, query):
+    location_prop_name = get_geo_case_property(domain)
+    case_ids = (
+        CaseSearchES()
+        .domain(domain)
+        .is_closed(False)
+        .search_string_query(query, ['name'])
+        .OR(
+            case_property_missing(location_prop_name),
+            case_property_query(location_prop_name, ""),
+        )
+        .sort('server_modified_on', desc=True)
+    ).get_ids()
+
+    paginator = Paginator(case_ids, limit)
+    case_ids_page = list(paginator.get_page(page))
+    cases = CommCareCase.objects.get_cases(case_ids_page, domain, ordered=True)
+    case_data = []
+    for c in cases:
+        case_data.append({
+            'id': c.case_id,
+            'name': c.name,
+        })
+    return {
+        'items': case_data,
+        'total': paginator.count,
+    }
+
+
+def _get_paginated_users_without_gps(domain, page, limit, query):
+    location_prop_name = get_geo_user_property(domain)
+    query = (
+        UserES()
+        .domain(domain)
+        .mobile_users()
+        .search_string_query(query, ['username'])
+        .sort('created_on', desc=True)
+    )
+
+    paginator = Paginator(query.get_ids(), limit)
+    user_ids_page = list(paginator.get_page(page))
+    user_docs = get_docs(CommCareUser.get_db(), keys=user_ids_page)
+    user_data = []
+    skipped_count = 0
+    for user_doc in user_docs:
+        if (
+            location_prop_name in user_doc['user_data']
+            and user_doc['user_data'][location_prop_name] != ''
+        ):
+            skipped_count += 1
```

Comment:
oh that is not going to be very effective. This would return different number of users in result for each page?

Reply:
I have updated this function to make use of new `UserES` filters, so `skipped_count` has been completely removed ([ff3d5d7](https://github.com/dimagi/commcare-hq/pull/33420/commits/ff3d5d7fa5fcd00a0fe805c83d8a48182bc43a08)).